### PR TITLE
MYR-7 : myrocks needs extra initialization step during .frm opening

### PR DESCRIPTION
**Pull request by george-lorch**
- Added virtual handler call init_with_fields, call from open_binary_frm and pass through partition handler as required by MyRocks and implemented in webscale.

### CODE REVIEW

**Comment by george-lorch** [comment]
https://blueprints.launchpad.net/percona-server/+spec/myrocks-extra-frm-init-5.6
https://jira.percona.com/browse/MYR-7
http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/1454/

**Comment by laurynas-biveinis** [comment]
It is not entirely clear why this needs to be done - MYR and blueprints say that it "needs", and that's all. A bit of suggestion is in the new field name and in the final diff comment re. HA_PRIMARY_KEY_IN_READ_INDEX - can you please elaborate?

**Comment by george-lorch** [comment]
The patch originated from Sergei Petrunia here https://github.com/facebook/mysql-5.6/commit/c1f1f0b64eaecd132fc6ec21663faf0510afacaf : Index-only scans do not work for partitioned tables and extended keys

I have added a bit more description in the MYR and blueprint taken from Sergeis original patch.

File: sql/ha_partition.h [hunk 1]
```diff
@@ -293,6 +293,14 @@ class ha_partition :public handler
                  ha_partition *clone_arg,
                  MEM_ROOT *clone_mem_root_arg);
    ~ha_partition();
+
+   /*
```
Comment (laurynas-biveinis) [inline sql/ha_partition.h]: This comment is redundant...

File: sql/handler.h [hunk 2]
```diff
@@ -2021,6 +2021,11 @@ class handler :public Sql_alloc
   {
     cached_table_flags= table_flags();
   }
+  /**
+    For MyRocks, secondary initialization that happens after frm is parsed into
```
Comment (laurynas-biveinis) [inline sql/handler.h:2025]: ... this one is not, and again it should mention as an example that MyRocks uses it to adjust HA_PRIMARY_KEY_IN_READ_INDEX

Comment (laurynas-biveinis) [reply sql/handler.h:2025]: Return values: false success, true failure

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by george-lorch** [comment]
Done...

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**